orangecontrib/spectroscopy/widgets/owfft.py

```diff
@@ -402,90 +403,116 @@ def calculateFFT(self):
                 peak_search=self.peak_search,
                 )
 
-        for row in ifg_data:
-            if self.sweeps in [2, 3]:
-                # split double-sweep for forward/backward
-                # forward: 2-2 = 0 , backward: 3-2 = 1
-                try:
-                    row = np.hsplit(row, 2)[self.sweeps - 2]
-                except ValueError as e:
-                    self.Error.ifg_split_error(e)
-                    return
+        if self.data.attributes is not None:
+            try:
+                self.reader = self.data.attributes['Reader']
+            except KeyError:
+                pass
+            
 
-            if self.sweeps in [0, 2, 3]:
-                try:
-                    spectrum_out, phase_out, wavenumbers = fft_single(
-                        row, zpd=stored_zpd_fwd, phase=stored_phase)
-                    zpd_fwd.append(fft_single.zpd)
-                except ValueError as e:
-                    self.Error.fft_error(e)
-                    return
-            elif self.sweeps == 1:
-                # Double sweep interferogram is split, solved independently and the
-                # two results are averaged.
-                try:
-                    data = np.hsplit(row, 2)
-                except ValueError as e:
-                    self.Error.ifg_split_error(e)
-                    return
+        if self.reader == 'NeaReaderGSF':
+            fft_single = irfft.ComplexFFT(dx=self.dx,
+                                 apod_func=self.apod_func,
+                                 zff=2**self.zff,
+                                 phase_res=self.phase_resolution if self.phase_res_limit else None,
+                                 phase_corr=self.phase_corr,
+                                 peak_search=self.peak_search,
+                                )
+            info = self.data.attributes
+            sh_data = self.data.X
+            full_data = sh_data[::2] * np.exp(sh_data[1::2]* 1j)
+
+            for row in full_data:
+                spectrum_out, phase_out, wavenumbers = fft_single.complex_fft(
+                                                            row, zpd=stored_zpd_fwd, info=info)
+                spectra.append(spectrum_out)
+                spectra.append(phase_out)
+            spectra = np.vstack(spectra)
```

Comment:
Why do you output interleaved amplitude/phase data instead of using the two outputs that already exist?

Reply:
To do this, I need to rebuild the data table, because the output is half the input, so I avoided that.
I think it's a good idea, but do I need to create a function to do this right?

Reply:
Okay let's leave input rows == output rows for now, we can correct to the 1 row per sample paradigm later.